google-cloud-video-live_stream/samples/acceptance/helper.rb

```diff
@@ -19,33 +19,146 @@
+    @channel_created_stopped = false
+    @input_created = false
+    @update_input_created = false
+    # Remove old channels and inputs in the test project if they exist
+    response = client.list_channels parent: location_path
+    response.each do |channel|
+      tmp = channel.name.to_s.split "-"
+      create_time = tmp.last.to_i
+      now = (Time.now.to_f * 1000).to_i # Kiloseconds, preserves float value for precision
+      next if create_time >= (now - DELETION_THRESHOLD_TIME_HOURS_IN_KILOSECONDS)
+      begin
+        operation = client.stop_channel name: channel.name.to_s
+        operation.wait_until_done!
+      rescue Google::Cloud::NotFoundError, Google::Cloud::FailedPreconditionError => e
+        puts "Rescued: #{e.inspect}"
+      end
+      begin
+        operation = client.delete_channel name: channel.name.to_s
+        operation.wait_until_done!
+      rescue Google::Cloud::NotFoundError => e
+        puts "Rescued: #{e.inspect}"
+      end
+    end
+
+    response = client.list_inputs parent: location_path
+    response.each do |input|
+      tmp = input.name.to_s.split "-"
+      create_time = tmp.last.to_i
+      now = (Time.now.to_f * 1000).to_i
+      next if create_time >= (now - DELETION_THRESHOLD_TIME_HOURS_IN_KILOSECONDS)
+      begin
+        operation = client.delete_input name: input.name.to_s
+        operation.wait_until_done!
+      rescue Google::Cloud::NotFoundError, Google::Cloud::FailedPreconditionError => e
+        puts "Rescued: #{e.inspect}"
+      end
+    end
+  end
 
   let :input do
     operation = client.create_input(
-      parent: input_location_path,
-      input:  {
-        type: Google::Cloud::Video::LiveStream::V1::Input::Type::RTMP_PUSH
-      },
+      parent: location_path,
+      input:  input_def,
       input_id: input_id
     )
     operation.wait_until_done!
     operation.response
   end
 
+  let :update_input do
+    operation = client.create_input(
+      parent: location_path,
+      input:  input_def,
+      input_id: update_input_id
+    )
+    operation.wait_until_done!
+    operation.response
+  end
+
+  let :channel do
+    input_path = client.input_path project: project_id, location: location_id, input: input_id
+    operation = client.create_channel parent: location_path, channel: channel_def(input_path, output_uri), channel_id: channel_id
+    operation.wait_until_done!
+    operation.response
+  end
+
+  let :started_channel do
+    input_path = client.input_path project: project_id, location: location_id, input: input_id
+    operation = client.create_channel parent: location_path, channel: channel_def(input_path, output_uri), channel_id: channel_id
+    operation.wait_until_done!
+    start = client.start_channel name: channel_name
+    start.wait_until_done!
+    operation.response
+  end
+
+
   after do
-    client.delete_input name: input_name
-  rescue Google::Cloud::NotFoundError
-    # Do nothing
+    if @channel_created_started == true
```

Comment:
You can just say `if @channel_created_started`. Same for the other if statements below.

Reply:
Done